pkg/summarizer/summary.go

```diff
@@ -486,53 +488,61 @@ func overallStatus(grid *statepb.Grid, recent int, stale string, alerts []*summa
 	return summarypb.DashboardTabSummary_UNKNOWN
 }
 
-func fmtStatus(passCols, cols, passCells, cells int) string {
-	colCent := 100 * float64(passCols) / float64(cols)
-	cellCent := 100 * float64(passCells) / float64(cells)
-	return fmt.Sprintf("%d of %d (%.1f%%) recent columns passed (%d of %d or %.1f%% cells)", passCols, cols, colCent, passCells, cells, cellCent)
-}
-
-func statusMessage(cols int, rows []*statepb.Row, recent int) string {
-	//  2483 of 115784 tests (2.1%) and 163 of 164 runs (99.4%) failed in the past 7 days
+// Culminate set of metrics related to a section of the Grid that can be consumed by
+func gridMetrics(cols int, rows []*statepb.Row, recent int) (int, int, int, int, []float32) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
 	results := results(ctx, rows)
+	var columnFailureRatios []float32
 	var passingCells int
 	var filledCells int
 	var passingCols int
 	var completedCols int
-	for idx := 0; idx < cols; idx++ {
-		if idx >= recent {
-			break
-		}
-		var passes bool
-		var failures bool
+
+	for idx := 0; idx < cols && idx < recent; idx++ {
+		var passes int
+		var failures int
 		for _, ch := range results {
 			// TODO(fejta): fail old running cols
 			switch coalesceResult(<-ch, result.IgnoreRunning) {
-			case statepb.Row_PASS:
-				if !failures {
-					passes = true
-				}
+			case statepb.Row_PASS, statepb.Row_PASS_WITH_ERRORS, statepb.Row_PASS_WITH_SKIPS:
+				passes++
 				passingCells++
 				filledCells++
 			case statepb.Row_NO_RESULT:
 				// noop
 			default:
```

Comment:
...for example, here, since there are technically some statuses that aren't failing OR passing, like flaky tests. (Doesn't need to be added this PR, but we should keep track of needing to add it).

Reply:
Yeah was not sure how to bin those, we could add metrics for that bin as well if we wanted and track a tuple of status... though not sure how much granularity we care about.

Reply:
I think we care less about 'granularity' than 'the concepts of passing and failing are consistent everywhere'. I don't think you have to do it this PR, but if you're interested we can sync some on doing that in a follow-up PR. ^^